docs/docsite/requirements.txt

```diff
@@ -1,8 +1,7 @@
-# pip packages required to build docsite
-# these requirements are as loosely defined as possible
-# if you want known good versions of these dependencies
-# use test/sanity/code-smell/docs-build.requirements.txt
-# instead
+# pip packages required to build the documentation.
+# These requirements are as loosely defined as possible.
+# For known good versions of doc build dependencies use the following command:
+# pip install --user -r tests/requirements.in -c tests/requirements.txt
 
 antsibull-docs >= 1.0.0, < 2.0.0
```

Comment:
What is the function of this separate file? Why not use the one in tests everywhere, or vice versa?
Also, these restrictions are in conflict with the test ones.

Reply:
Yes I noticed the conflicting restrictions too but didn't want to start changing things here and changing the scope of this PR.

The function, as I understand it, is to have a requirements file that users can modify as they like and experiment with different requirements etc. Although I don't see why a separate file is necessary to do that. Perhaps someone like @felixfontein or @samccann could give a little more background because this has been around for a while and I'm not sure who in the community might be using it.

Reply:
Why don't we just replace this with the new requirements file? If users want the unconstrained deps, they can use requirements.in without the pins in requirements.txt. We can add a symlink from one location to another if that's needed for whatever reason. Having two sets of conflicting requirements files does not make sense to me.

Reply:
It doesn't make sense to me either to have conflicting requirements files but I would prefer to handle it as a separate issue. Using `requirements.in` is a great suggestion but I wonder if that might change how @felixfontein and possibly others use `docs/docsite/requirements.txt`.

Reply:
Yeah, i'd drop one of these but in a follow-up to scope the current effort better.

Reply:
I don't particularly want to merge this with these conflicting requirements, but they were there before, so I guess...

Reply:
You're spot on with that docs update @gotmax23 Thanks for the suggestion!

Reply:
I only use the requirements files when I update them in PRs to bump the antsibull-docs/... version :) I never used them to install anything (except for looking inside to see what is installed, and installing the missing pieces manually).

Reply:
> Why don't we just replace this with the new requirements file? If users want the unconstrained deps, they can use requirements.in without the pins in requirements.txt. We can add a symlink from one location to another if that's needed for whatever reason. Having two sets of conflicting requirements files does not make sense to me.

:+1: for that. I would probably have the main requirement files in `docs/`, and symlinks from `tests/requirements.txt` to `docs/requirements.txt` and `docs/docsite/requirements.txt` to `docs/requirements.in`.

Reply:
The reason for two files:
`/docs/requirements.txt` was supposed to be as open as possible and used to publish to devel. The idea was to catch any problems with requirements long before we wanted to update the stable requirements (reflected in `tests/requirements.txt`).
`test/requirements.txt` - to be the sable, pinned set of requirements we use to publish to latest so we know the docs will build, and also be identical to the set of requirements used by CI.

Every so often, we would go into the stable one, update requirements to more recent versions, test, and pin at those versions ...and use that set for another few months.

These were not intended for users to experiment with, tho our docs local build instructions say they can use either set.

We could get rid of the open set of requirements I suppose, but that leaves us unable for example, to test antsibull-docs version 'live' on devel for a soak period before enabling that same version on latest. 

My high-level objective is to always have stable requirements for our latest release. That means it has to be pinned.

Reply:
> We could get rid of the open set of requirements I suppose, but that leaves us unable for example, to test antsibull-docs version 'live' on devel for a soak period before enabling that same version on latest.

@samccann that's not really true — it is possible to just not pass `--constraint` and use the unpinned stuff if needed. But I'd recommend getting rid of two versions of unpinned deps lists and have just one.

Reply:
> My high-level objective is to always have stable requirements for our latest release. That means it has to be pinned.

Yes, and when the pins are updated (by `pip-tools` in our case), this is performed through PRs and the build runs with those, so CI is able to catch any problems and what's more important — every dep bump is versioned in Git making it easier to track when things start to misbehave.